usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedCalendarTest.java

```diff
@@ -0,0 +1,250 @@
+
+        when(mockGenericObject.getData()).thenReturn(mockCalendar);
+        when(mockGenericObject.isSuccess()).thenReturn(true);
+
+        //noinspection unchecked
+        when(mockBuilder.build()).thenReturn(mockGenericObject);
+
+        when(mockParsedCalendar.get(SERVICE_ID)).thenReturn(SERVICE_ID);
+        when(mockParsedCalendar.get(MONDAY)).thenReturn(0);
+        when(mockParsedCalendar.get(TUESDAY)).thenReturn(0);
+        when(mockParsedCalendar.get(WEDNESDAY)).thenReturn(0);
+        when(mockParsedCalendar.get(THURSDAY)).thenReturn(0);
+        when(mockParsedCalendar.get(FRIDAY)).thenReturn(0);
+        when(mockParsedCalendar.get(SATURDAY)).thenReturn(0);
+        when(mockParsedCalendar.get(SUNDAY)).thenReturn(0);
+        when(mockParsedCalendar.get(START_DATE)).thenReturn(LocalDateTime.now());
+        when(mockParsedCalendar.get(END_DATE)).thenReturn(LocalDateTime.now());
+
+        when(mockGtfsDataRepo.addCalendar(mockCalendar)).thenReturn(mockCalendar);
+
+        final ProcessParsedCalendar underTest = new ProcessParsedCalendar(mockResultRepo, mockGtfsDataRepo, mockBuilder);
+
+        underTest.execute(mockParsedCalendar);
+
+        verify(mockParsedCalendar, times(1)).get(SERVICE_ID);
+        verify(mockParsedCalendar, times(1)).get(MONDAY);
+        verify(mockParsedCalendar, times(1)).get(TUESDAY);
+        verify(mockParsedCalendar, times(1)).get(WEDNESDAY);
+        verify(mockParsedCalendar, times(1)).get(THURSDAY);
+        verify(mockParsedCalendar, times(1)).get(FRIDAY);
+        verify(mockParsedCalendar, times(1)).get(SATURDAY);
+        verify(mockParsedCalendar, times(1)).get(SUNDAY);
+        verify(mockParsedCalendar, times(1)).get(START_DATE);
+        verify(mockParsedCalendar, times(1)).get(END_DATE);
+
+        verify(mockBuilder, times(1)).monday(ArgumentMatchers.eq(0));
+        verify(mockBuilder, times(1)).tuesday(ArgumentMatchers.eq(0));
+        verify(mockBuilder, times(1)).wednesday(ArgumentMatchers.eq(0));
+        verify(mockBuilder, times(1)).thursday(ArgumentMatchers.eq(0));
+        verify(mockBuilder, times(1)).friday(ArgumentMatchers.eq(0));
+        verify(mockBuilder, times(1)).saturday(ArgumentMatchers.eq(0));
+        verify(mockBuilder, times(1)).sunday(ArgumentMatchers.eq(0));
+        verify(mockBuilder, times(1)).serviceId(SERVICE_ID);
+        verify(mockBuilder, times(1)).startDate(ArgumentMatchers.isA(LocalDateTime.class));
+        verify(mockBuilder, times(1)).endDate(ArgumentMatchers.isA(LocalDateTime.class));
+
+        final InOrder inOrder = inOrder(mockBuilder, mockGtfsDataRepo);
+
+        inOrder.verify(mockBuilder, times(1)).build();
+
+        inOrder.verify(mockGtfsDataRepo, times(1)).addCalendar(ArgumentMatchers.eq(mockCalendar));
+
+        verifyNoMoreInteractions(mockBuilder, mockCalendar, mockParsedCalendar, mockGtfsDataRepo);
+    }
+
+    @Test
+    public void invalidCalendarShouldAddNoticeToResultRepoAndShouldNotBeAddedToGtfsDataRepo() {
```

Comment:
both those tests should be refactored as one testing for add rejection by the repository, no matter the reason

Reply:
Nope, the duplication notice is generated inside the use case, so it makes sens it would be tested here